src/Pathfinding/ObstacleMap.cpp

```diff
@@ -34,18 +20,19 @@ inline bool inBounds(int coordinate)
 }
 
 //rebuilds ObstacleMap with given Obstacles
-void ObstacleMap::update(std::vector<PointXY>& obstacles)
+void ObstacleMap::update(const std::vector<PointXY>& obstacles, float robotX, float robotY)
 {
-    updateRobotPosition();//updates values in robotX and robotY fields
+    this->robotX = robotX;
+    this->robotY = robotY;
```

Comment:
Minor quibble: it's slightly inconsistent that we use `this->robotX` here and just `robotX` below